eventsourcing/src/test/java/org/axonframework/eventsourcing/eventstore/jdbc/MysqlJdbcEventStorageEngineTest.java

```diff
@@ -36,56 +37,58 @@
  *
  * @author Albert Attard (JavaCreed)
  */
+@Testcontainers
 class MysqlJdbcEventStorageEngineTest {
 
-    private MysqlDataSource dataSource;
+    @Container
+    private static final MySQLContainer<?> MYSQL_CONTAINER = new MySQLContainer<>("mysql")
+            .withDatabaseName("axon")
+            .withUsername("admin")
+            .withPassword("some-password");
+
     private JdbcEventStorageEngine testSubject;
 
     @BeforeEach
-    void setUp() throws Exception {
-        /* Load the DB properties */
-        final Properties properties = new Properties();
-        properties.load(getClass().getResourceAsStream("/mysql.test.database.properties"));
-
-        dataSource = new MysqlDataSource();
-        dataSource.setUrl(properties.getProperty("jdbc.url"));
-        dataSource.setUser(properties.getProperty("jdbc.username"));
-        dataSource.setPassword(properties.getProperty("jdbc.password"));
+    void setUp() {
+        MysqlDataSource dataSource = new MysqlDataSource();
+        dataSource.setUrl(MYSQL_CONTAINER.getJdbcUrl());
+        dataSource.setUser(MYSQL_CONTAINER.getUsername());
+        dataSource.setPassword(MYSQL_CONTAINER.getPassword());
         try {
-            testSubject = createEngine(new SQLErrorCodesResolver(dataSource));
+            testSubject = createEngine(dataSource);
         } catch (Exception e) {
             throw new TestAbortedException("Ignoring this test, as no valid MySQL instance is configured", e);
```

Comment:
With current approach we should be able to remove this try-catch block, right?
Same applies for the other 2 tests.

Reply:
Sure, let's put some more trust in test containers!